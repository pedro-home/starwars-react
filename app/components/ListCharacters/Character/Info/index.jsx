import React from 'react';
import { FlexboxGrid } from 'rsuite';
import PropTypes from 'prop-types';

const Component = ({ type, value }) => {
  return (
    <FlexboxGrid justify="space-between" className="character-info">
      <FlexboxGrid.Item>
        <span className="character-info-type">{type}</span>
      </FlexboxGrid.Item>
      <FlexboxGrid.Item>
        <span className="character-info-value">{value}</span>
      </FlexboxGrid.Item>
    </FlexboxGrid>
  );
};

Component.propTypes = {
  type: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
};

export default Component;
