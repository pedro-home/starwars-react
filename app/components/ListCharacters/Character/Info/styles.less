.character-info {
  text-transform: capitalize;

  &-type {
    color: @base-color;
  }
}
