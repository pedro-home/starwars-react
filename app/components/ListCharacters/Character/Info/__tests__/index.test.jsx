import React from 'react';
import { render } from '@testing-library/react';

import Info from '../index';

describe('Given <Info />', () => {
  let component;
  let defaultProps;

  beforeAll(() => {
    defaultProps = { type: 'hello', value: 'world' };
  });

  describe('When initializes', () => {
    beforeEach(() => {
      const { container } = render(<Info {...defaultProps} />);
      component = container;
    });

    it('Then should match the snapshot', () => {
      expect(component).toMatchSnapshot();
    });

    it('And should render a type', () => {
      expect(component.querySelector('.character-info-type').innerHTML).toEqual(defaultProps.type);
    });

    it('And should render a value', () => {
      expect(component.querySelector('.character-info-value').innerHTML).toEqual(defaultProps.value);
    });
  });
});
