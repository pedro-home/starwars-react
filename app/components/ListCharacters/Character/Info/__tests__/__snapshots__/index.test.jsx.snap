// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <Info /> When initializes Then should match the snapshot 1`] = `
<div>
  <div
    class="rs-flex-box-grid character-info rs-flex-box-grid-top rs-flex-box-grid-space-between"
  >
    <div
      class="rs-flex-box-grid-item rs-flex-box-grid-item-0"
    >
      <span
        class="character-info-type"
      >
        hello
      </span>
    </div>
    <div
      class="rs-flex-box-grid-item rs-flex-box-grid-item-0"
    >
      <span
        class="character-info-value"
      >
        world
      </span>
    </div>
  </div>
</div>
`;
